src/walletSdk/horizon/Account.ts

```diff
@@ -0,0 +1,52 @@
+import { Keypair, Transaction, FeeBumpTransaction } from "stellar-sdk";
+
+class AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    this.keypair = keypair;
+  }
+
+  get publicKey(): string {
+    return this.keypair.publicKey();
+  }
+
+  toString(): string {
+    return `${this.constructor.name}(address=${this.publicKey})`;
+  }
+}
+
+export class PublicKeypair extends AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    super(keypair);
+  }
+}
+
+export class SigningKeypair extends AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    if (!keypair.canSign()) {
+      throw new Error("This keypair doesn't have a private key and can't sign");
+    }
+    super(keypair);
+  }
+
+  static fromSecret = (secret: string): SigningKeypair => {
+    return new SigningKeypair(Keypair.fromSecret(secret));
+  };
+
+  get secretKey(): string {
+    return this.keypair.secret();
+  }
+
+  sign(
```

Comment:
I played around with extending the stellar-sdk Transaction class using augmentation in Typescript (to make similar to [kotlin here](https://github.com/stellar/kotlin-wallet-sdk/blob/e6c801e24e7c716b876a67098131dd0b4d2f1691/wallet-sdk/src/main/kotlin/org/stellar/walletsdk/horizon/Account.kt#L46)). It gets pretty messy imo. So I left just the `sign` method here for signing with keypair

to sign with the transaction someone could just do: `tx.sign(signingKeypair.keypair)`

Reply:
Yeah I'm guessing all extensions functions we have on kotlin side should either be a part of original class, or if it extends some external functionality you can just use util functions